tests/validator/test_validation_graph.py

```diff
@@ -185,3 +197,279 @@ def test_ExpectationValidationGraph_get_exception_info(
 
     assert left_exception in exception_info
     assert right_exception in exception_info
+
+
+@pytest.mark.integration
+def test_parse_validation_graph():
+    class PandasExecutionEngineStub:
+        pass
+
+    PandasExecutionEngineStub.__name__ = "PandasExecutionEngine"
+    pandas_execution_engine_stub = cast(ExecutionEngine, PandasExecutionEngineStub())
+
+    expectation_configuration = ExpectationConfiguration(
+        expectation_type="expect_column_value_z_scores_to_be_less_than",
+        kwargs={
+            "column": "a",
+            "mostly": 0.9,
+            "threshold": 4,
+            "double_sided": True,
+        },
+    )
+    graph = ValidationGraph(execution_engine=pandas_execution_engine_stub)
+    validation_dependencies = (
+        ExpectColumnValueZScoresToBeLessThan().get_validation_dependencies(
+            expectation_configuration, pandas_execution_engine_stub
+        )
+    )
+
+    for metric_configuration in validation_dependencies["metrics"].values():
+        graph.build_metric_dependency_graph(
+            metric_configuration=metric_configuration,
+            runtime_configuration=None,
+        )
+
+    ready_metrics, needed_metrics = graph._parse(metrics=dict())
+    assert len(ready_metrics) == 2 and len(needed_metrics) == 9
```

Comment:
This seems like the part you want to test based on the test name. The part above seems like setup that is repeated in a few tests. Could you move that out into its own helper setup function that each of the tests call?

Reply:
@billdirks Updated -- great idea -- thanks!